src/Persistence.LinnApps/Repositories/StorageLocationRepository.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace Linn.Stores2.Persistence.LinnApps.Repositories
+{
+    using System.Threading.Tasks;
+
+    using Microsoft.EntityFrameworkCore;
+    using Linn.Common.Persistence.EntityFramework;
+    using Linn.Stores2.Domain.LinnApps.Stock;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System;
+
+    public class StorageLocationRepository : EntityFrameworkRepository<StorageLocation, int>
+    {
+        private readonly ServiceDbContext serviceDbContext;
+
+        public StorageLocationRepository(ServiceDbContext serviceDbContext) : base(serviceDbContext.StorageLocations)
+        {
+            this.serviceDbContext = serviceDbContext;
+        }
+
+
+        public override IQueryable<StorageLocation> FindAll()
+        {
+            // dont need every old location
+            return this.serviceDbContext.StorageLocations.Where(l => l.DateInvalid == null).OrderBy(l => l.LocationCode);
```

Comment:
Usually don't apply filters in the repo .FindAll(), but if you think there will never be a need to include old ones I guess it won't hurt

Reply:
i don't actually think you ever really want to FindAll() at all given it is alot and includes the sheer number of locations including entire buildings we no longer own